### PR TITLE
app: Fix clusters delegates table check

### LINKED ISSUE #25984: Various SetDefaultDelegate/GetDelegate cluster implementations are broken for dynamic endpoints
For example, window covering has:

```
Delegate * GetDelegate(EndpointId endpoint)
{
    uint16_t ep = emberAfFindClusterServerEndpointIndex(endpoint, WindowCovering::Id);
    return ((ep == kInvalidEndpointId || ep >= EMBER_AF_WINDOW_COVERING_CLUSTER_SERVER_ENDPOINT_COUNT) ? nullptr
                                                                                                       : gDelegateTable[ep]);
}
```

which will always return null if `endpoint` is a dynamic endpoint.  It also has:

```
void SetDefaultDelegate(EndpointId endpoint, Delegate * delegate)
{
    uint16_t ep = emberAfFindClusterServerEndpointIndex(endpoint, WindowCovering::Id);

    // if endpoint is found and is not a dynamic endpoint
    if (ep != 0xFFFF && ep < EMBER_AF_WINDOW_COVERING_CLUSTER_SERVER_ENDPOINT_COUNT)
```

which will likewise fail for dynamic endpoints.  The fact that `gDelegateTable` is sized large enough for static + dynamic endpoints does not help.

Just changing the bounds checks does not really resolve the problem, because the code above relies on endpoints never changing their position in "the list of endpoints that implement the given cluster", but with dynamic endpoints they can absolutely change their position.

What we probably need to do is that if the endpoint is static we use `emberAfFindClusterServerEndpointIndex` and otherwise we just `endpoint index - total number of static endpoints + EMBER_AF_WINDOW_COVERING_CLUSTER_SERVER_ENDPOINT_COUNT` or something along those lines, to guarantee index stability.

This applies to a number of clusters:
```
 % git grep -l SetDefaultDelegate src/app/clusters/*/*.cpp
src/app/clusters/account-login-server/account-login-server.cpp
src/app/clusters/application-basic-server/application-basic-server.cpp
src/app/clusters/application-launcher-server/application-launcher-server.cpp
src/app/clusters/audio-output-server/audio-output-server.cpp
src/app/clusters/channel-server/channel-server.cpp
src/app/clusters/content-launch-server/content-launch-server.cpp
src/app/clusters/keypad-input-server/keypad-input-server.cpp
src/app/clusters/low-power-server/low-power-server.cpp
src/app/clusters/media-input-server/media-input-server.cpp
src/app/clusters/media-playback-server/media-playback-server.cpp
src/app/clusters/target-navigator-server/target-navigator-server.cpp
src/app/clusters/wake-on-lan-server/wake-on-lan-server.cpp
src/app/clusters/window-covering-server/window-covering-server.cpp
```

In practice that means Window Covering and the media clusters.

### LINKED ISSUE #26959: Color Control server is broken with dynamic endpoints
Similar to https://github.com/project-chip/connectedhomeip/issues/25984 but for Color Control server, which does:

```
    uint16_t index            = emberAfFindClusterServerEndpointIndex(endpoint, ColorControl::Id);
    EmberEventControl * event = nullptr;

    if (index < ArraySize(eventControls))
```

where 

```
    EmberEventControl eventControls[EMBER_AF_COLOR_CONTROL_CLUSTER_SERVER_ENDPOINT_COUNT];
```

@jmartinez-silabs

### PR DESCRIPTION
**Pull request by wqx6**
We should use table size instead of the fixed cluster endpoint count for the check for the cluster servers delegate.

### CODE REVIEW

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]
This is not going to help on its own, because for dynamic endpoints emberAfFindClusterServerEndpointIndex will return garbage.

You need to use `emberAfGetClusterServerEndpointIndex` instead at all these callsites...

**Comment by wqx6** [comment]
@bzbarsky-apple Replaced all the emberFindClusterServerEndpointIndex now

File: src/app/clusters/account-login-server/account-login-server.cpp [hunk 1]
```diff
@@ -66,8 +66,9 @@ Delegate * GetDelegate(EndpointId endpoint)
 #endif // CHIP_DEVICE_CONFIG_APP_PLATFORM_ENABLED
     ChipLogProgress(Zcl, "AccountLogin NOT returning ContentApp delegate for endpoint:%u", endpoint);
 
-    uint16_t ep = emberAfFindClusterServerEndpointIndex(endpoint, AccountLogin::Id);
-    return ((ep == 0xFFFF || ep >= EMBER_AF_ACCOUNT_LOGIN_CLUSTER_SERVER_ENDPOINT_COUNT) ? nullptr : gDelegateTable[ep]);
+    uint16_t ep =
+        emberAfGetClusterServerEndpointIndex(endpoint, AccountLogin::Id, EMBER_AF_ACCOUNT_LOGIN_CLUSTER_SERVER_ENDPOINT_COUNT);
+    return ((ep == 0xFFFF || ep >= kAccountLoginDeletageTableSize) ? nullptr : gDelegateTable[ep]);
```
Comment (bzbarsky-apple) [inline src/app/clusters/account-login-server/account-login-server.cpp:71]: Please use kEmberInvalidEndpointIndex, not hardcoded numbers. Same at all the other callsites.

That said, in practice, isn't the check for kEmberInvalidEndpointIndex redundant with the comparison to table size?

**Review by bzbarsky-apple — APPROVED** [review]

File: src/app/clusters/account-login-server/account-login-server.cpp (hunk 1 above)
Comment (wqx6) [reply src/app/clusters/account-login-server/account-login-server.cpp]: Remove the InvalidEndpointIndex check now.

**Review by jmartinez-silabs — APPROVED** [review]
~~I was actually doing this too.~~
~~Can you remove `emberAfFindClusterServerEndpointIndex` altogether?~~
~~I believe the only place left that uses it is in the tv-app and in a mock for Darwin tests.~~

~~after that there is no more use for `emberAfFindClusterServerEndpointIndex`~~

Since this PR already pass CI and all, ill just merge it and I can do the deletion in another PR

**Pull request merged**